presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -137,9 +140,25 @@ BroadcastFileReader::BroadcastFileReader(
     velox::memory::MemoryPool* pool)
     : broadcastFileInfo_(std::move(broadcastFileInfo)),
       fileSystem_(fileSystem),
+      asyncDataCache_(velox::cache::AsyncDataCache::getInstance()),
       hasData_(true),
       numBytes_(0),
-      pool_(pool) {}
+      lease_(fileIds(), broadcastFileInfo_->filePath_),
+      pool_(pool) {
+  auto readFile = fileSystem_->openFileForRead(broadcastFileInfo_->filePath_);
+  numBytes_ = readFile->size();
+
+  bufferedInput_ = std::make_unique<velox::dwio::common::CachedBufferedInput>(
+      std::move(readFile),
+      velox::dwio::common::MetricsLog::voidLog(),
```

Comment:
Hmm... this reference to dwio suggests that some refactoring is needed to allow CachedBufferedInput to be used outside of TableScan. Let's do that before starting to use it here.